### PR TITLE
Change uavcan_debug char* args to const

### PR DESCRIPTION
**Pull request by gwenz-matternet**
- All boards in station_firmwares repo compile ok with this change
- I did not try to compile any other projects

### CODE REVIEW

**Comment by gwenz-matternet** [comment]
@afbowen @rabad-matternet Is there any reason we might *not* want to change these args to `const`?

**Comment by rabad-matternet** [comment]
> @afbowen @rabad-matternet Is there any reason we might _not_ want to change these args to `const`?

No, I see no reason why we might not want to change these to const.  In other words, I agree with changing the declaration to const.

It can be even further argued that if you want to protect the integrity of the input being passed to these functions that you may want to declare it as:

const char* const arg

Though this isn't necessary in these functions as they do not attempt to reassign args (i.e. source and key).

File: modules/uavcan_debug/uavcan_debug.c [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #include <chprintf.h>
 #include <memstreams.h>
 
-void uavcan_send_debug_msg(uint8_t debug_level, char* source, const char *fmt, ...)
+void uavcan_send_debug_msg(uint8_t debug_level, const char* source, const char *fmt, ...)
```
Comment (rabad-matternet) [inline modules/uavcan_debug/uavcan_debug.c]: according to our style guide:

const char *fmt

should be:

const char* fmt

File: modules/uavcan_debug/uavcan_debug.h [hunk 2]
```diff
@@ -8,5 +8,5 @@
 #define LOG_LEVEL_WARNING UAVCAN_PROTOCOL_DEBUG_LOGLEVEL_WARNING
 #define LOG_LEVEL_ERROR UAVCAN_PROTOCOL_DEBUG_LOGLEVEL_ERROR
 
-void uavcan_send_debug_msg(uint8_t debug_level, char* source, const char *fmt, ...);
-void uavcan_send_debug_keyvalue(char* key, float value);
+void uavcan_send_debug_msg(uint8_t debug_level, const char* source, const char *fmt, ...);
```
Comment (rabad-matternet) [inline modules/uavcan_debug/uavcan_debug.h]: same comment as above in regards to style guide

File: modules/uavcan_debug/uavcan_debug.c (hunk 1 above)
Comment (gwenz-matternet) [reply modules/uavcan_debug/uavcan_debug.c]: Technically, we only have a C++ style guide right now. :-)
- I agree with your suggestion tho, and I have changed it.
- I would like for our C++ and C style to be as similar as possible, though there will be some areas where I think they'll have to diverge.
- Hopefully soon we will be able to articulate an Embedded C style guide for the company.

**Comment by gwenz-matternet** [comment]
> It can be even further argued
>
> const char* const arg

Good point. I'm slightly inclined to leave it as-is, because:
1. It's less information to absorb when reading the code.
2. It affects only a very small amount of code, and doesn't affect the API itself.

I should note that the reason I am changing is that I have a `const char*` that I want to pass in from my application code, and compiler does not (and should not!) let you pass `const char*` to a function that only takes `char *`.

**Comment by gwenz-matternet** [comment]
@afbowen In case you haven't come come across this in your embedded C learning, there is a "read backwards" trick that is helpful to make sense of C language's confusing declaration syntax:
https://stackoverflow.com/questions/1143262/what-is-the-difference-between-const-int-const-int-const-and-int-const

From the above, I also just came across the more general "Clockwise/Spiral Rule": 🤔 
http://c-faq.com/decl/spiral.anderson.html

**Review by afbowen — APPROVED** [review]
Surprised this wasn't already a `const` argument, thanks for updating!

**Review by rabad-matternet — APPROVED** [review]

**Pull request merged**